parser/parser_test.go

```diff
@@ -287,6 +284,23 @@ func (s *testParserSuite) TestParser0(c *C) {
 		}
 	}
 
+	// Testcase for unreserved keywords
+	unreservedKws := []string{
+		"auto_increment", "after", "begin", "bit", "bool", "boolean", "charset", "columns", "commit",
+		"date", "datetime", "deallocate", "do", "end", "engine", "engines", "execute", "first", "full",
+		"local", "names", "offset", "password", "prepare", "quick", "rollback", "session", "signed",
+		"start", "global", "tables", "text", "time", "timestamp", "transaction", "truncate", "unknown",
+		"value", "warnings", "year", "now", "substring", "mode",
+	}
+	for _, kw := range unreservedKws {
+		src := fmt.Sprintf("SELECT %s FROM tbl;", kw)
+		fmt.Printf("%s\n", src)
```

Comment:
You can pass `Commentf(src)` as the last argument in `c.Assert` method to do that.